### PR TITLE
PDX-349: Add /api/states API

### PR DESCRIPTION
**Pull request by XxshiftxX**

### CODE REVIEW

File: EmptyChronicle/Application/States/StatesApplication.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        StatesRepository = statesRepository;
+    }
+
+    public State? GetStateByAddress(string address, string? accountAddress = null)
+    {
+        return accountAddress is null
+            ? StatesRepository.GetStateByAddress(address)
+            : StatesRepository.GetStateByAddress(address, accountAddress);
+    }
+}
```
Comment (moreal) [inline EmptyChronicle/Application/States/StatesApplication.cs:20]: Please insert newline at the EOF, for all changed files in this pull request.

File: EmptyChronicle/Controller/StatesController.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    public StatesController(StatesApplication statesApplication)
+    {
+        StatesApplication = statesApplication;
+    }
+
+    [HttpGet("{address}")]
+    public ActionResult GetStatesByAddress(string address, [FromQuery(Name = "account")] string? accountAddress)
+    {
+        var state = StatesApplication.GetStateByAddress(address, accountAddress);
```
Comment (moreal) [inline EmptyChronicle/Controller/StatesController.cs:21]: It should return `BadRequest()` when the `StatesApplication.GetStateByAddress` throws `ArgumentException` by [`Libplanet.Address.#ctor(string)`](https://docs.libplanet.io/4.0.6/api/Libplanet.Crypto.Address.html#Libplanet_Crypto_Address__ctor_System_String_).

File: EmptyChronicle/Controller/BlockController.cs [hunk 3]
```diff
@@ -71,4 +71,4 @@ public BlockDto GenerateBlockDto(Block block)
             }).ToArray(),
         };
     }
-}
+}
```
Comment (moreal) [inline EmptyChronicle/Controller/BlockController.cs:74]: It seems like unintended changes. You should check your IDE setup whether it removes all newline at the EOF by default. 🤔

**Review by moreal — CHANGES_REQUESTED** [review]

File: EmptyChronicle/Program.cs [hunk 4]
```diff
@@ -17,7 +17,7 @@ bool odin
         (true, true) => throw new CommandExitedException("You must select only one of the --heimdall and --odin options.", -1),
         (true, false) => "appsettings.heimdall.json",
         (false, true) => "appsettings.odin.json",
-        (false, false) => throw new CommandExitedException("You must choose between the --heimdall option and the --odin option.", -1)
+        (false, false) => "appsettings.heimdall.json"
```
Comment (moreal) [inline EmptyChronicle/Program.cs:20]: Why was it changed?

Comment (XxshiftxX) [reply EmptyChronicle/Program.cs:20]: Oh it just my personal settings so I will rollback it soon. Thank you for notice.

**Review by moreal — APPROVED** [review]

**Pull request merged**